mysticeti-core/src/syncer.rs

```diff
@@ -50,8 +51,13 @@ impl<H: BlockHandler, S: SyncerSignals, C: CommitObserver> Syncer<H, S, C> {
             .metrics
             .utilization_timer
             .utilization_timer("Syncer::add_blocks");
-        self.core.add_blocks(blocks);
-        self.try_new_block(connected_authorities);
+        for block in blocks
+            .into_iter()
+            .sorted_by(|b1, b2| b1.round().cmp(&b2.round()))
+        {
+            self.core.add_blocks(vec![block]);
+            self.try_new_block(connected_authorities.clone());
```

Comment:
This would mean when we catching up, we would create new block every round we missed. This could hurt performance quite a bit is not it?